contracts/gateway/GatewayBase.sol

```diff
@@ -257,6 +273,8 @@ contract GatewayBase is Organized {
         );
 
         storageRoots[_blockHeight] = storageRoot;
+        uint256 oldestStoredBlockHeight = CircularBufferUintLib.store(_blockHeight, storageRootBuffer);
+        delete storageRoots[oldestStoredBlockHeight];
```

Comment:
I see why have you returned the previously stored element from the circular buffer store function.
What do you think about storing both, blockHeight and storageRoot in circular buffer itself. It would simplify code and make it more robust. Thoughts?

Reply:
If we store both blockHeight and storeageRoot in the circular buffer then, it will become very specific implementation. Currently the circular buffer is generic and can be used for any data other than blockHeight and storageRoot.